Extending/Packages/Good-practice/index.md

```diff
@@ -0,0 +1,141 @@
+
+As the files will still be copied during build you should ensure your target file doesn't overwrite newer or altered files and doesn't delete files on clean. 
+
+_This method isn't perfect and if you do have a package that is heavy on views you might consider building a dotnet template or Razor class library to contain the views so they don't pollute your user's solutions._ 
+
+## License files
+umbraco products store thier licenses in `/umbraco/Licences` and it is considered a goog location for third party packages that require license files to also store their licence files there. 
+
+_The default .gitignore for Umbraco templates will include any files in this folder in a repo (while ignoring most of the rest of the Umbraco folder_
+
+:::note
+The  `/umbraco/Licenses` folder does not exist on a fresh installation of Umbraco so you might need to create it before you save your license file to this folder. 
+:::
+
+## Operating system considerations
+Umbraco 9 is a .NET Core application and it can be ran on multiple operating systems (e.g Windows, Linux). When developing packages there are a few things you should be aware of so your package can run on all possible operating systems. 
+
+### Case sensitivity 
+The Linux file system is case sensitive, so `App_Plugins/myPackage` is a different location to `app_plugins/MyPACKAGE`. When building your package you should ensure that you always refer to folders and paths in a consistent way. 
+
+:::tip
+You can adjust the case sensitivity of a Windows folder by running a command against the folder. 
+
+``` bash
+fsutil.exe file queryCaseSensitiveInfo <path>
+```
+:::
+
+### Well known folders and case sensitivity  
+Some folders within Umbraco will already exist for all installations, if you access these folders you need to be aware of the case used to ensure you end up in the right place: 
+
+| Folder | Note | 
+|-|-|
+| /App_Plugins | Uppercase `A` and `P` |
+| /App_Pluigns/[Ll]ang | Uppercase `L` but in later version of Umbraco 9.3+ can be upper or lowercase `L` |
+| /Views | Uppercase `V` | 
+| /umbraco/Licenses | Lowercase `u` uppercase `L` |
+| /config | Lowercase `c` |
+
+:::note 
+If you create a custom section/tree umbraco will also build paths based on the name of that section or tree, these folder paths will also be case sensitive: 
+
+For example If you have a custom tree with the treeAlias of `MyCustomTree` umbraco will look for files in `App_Plugins\MyCustomTree\backoffice\MyCustomTree\` 
+:::
+
+### Path manipulation
+Building folder path strings manually can cause problems when swapping between file systems. Windows uses the backslash character to '\' separate folders and files, Linux uses the forward slash '/'.
+
+You should use the .Net `Path` commands wherever possible when building paths to ensure that the correct path is build. 
+
+``` cs
+// WRONG: don't build paths manually.
+var myPath = myFolder + "\myfile.txt";
```

Comment:
I think i need to add something about getting paths via the FileProvider or `MapPath` commands, because it should never be hardwired, but this example could then have a mapped path value for 'myFolder' and it would make more sense.